catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/CswTransformProvider.java

```diff
@@ -112,12 +112,16 @@ public void marshal(Object o, HierarchicalStreamWriter writer, MarshallingContex
   private void writeXml(BinaryContent content, HierarchicalStreamWriter writer) {
     try {
       XmlPullParser parser = XppFactory.createDefaultParser();
-      new HierarchicalStreamCopier()
-          .copy(
-              new XppReader(
-                  new InputStreamReader(content.getInputStream(), StandardCharsets.UTF_8), parser),
-              writer);
-    } catch (XmlPullParserException e) {
+      InputStreamReader inputStreamReader =
+          new InputStreamReader(content.getInputStream(), StandardCharsets.UTF_8);
+      XppReader xppReader = new XppReader(inputStreamReader, parser);
+      try {
+        new HierarchicalStreamCopier().copy(xppReader, writer);
+      } finally {
+        inputStreamReader.close();
+        xppReader.close();
```

Comment:
❗️ The `xppReader` can be left open if closing the stream reader throws an exception. Either use `IOUtils.closeQuietly` or wrap the close in a try/catch.